src/integer_mod_q/mat_polynomial_ring_zq/from.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright © 2023 Marcel Luca Schmidt
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! Implementations to create a [`MatPolynomialRingZq`] value from other types.
+//! For each reasonable type, an explicit function with the format
+//! `from_<type_name>` and the [`From`] trait should be implemented.
+//!
+//! The explicit functions contain the documentation.
+
+use std::fmt::Display;
+
+use super::MatPolynomialRingZq;
+use crate::{error::MathError, integer::MatPolyOverZ, integer_mod_q::ModulusPolynomialRingZq};
+
+impl MatPolynomialRingZq {
+    /// Creates a new matrix with `num_rows` rows, `num_cols` columns,
+    /// zeros as entries and `modulus` as the modulus.
+    ///
+    /// Parameters:
+    /// - `num_rows`: number of rows the new matrix should have
+    /// - `num_cols`: number of columns the new matrix should have
+    /// - `modulus`: the common modulus of the matrix entries
+    ///
+    /// Returns a [`MatPolynomialRingZq`] or an error, if the number of rows or columns is
+    /// less than `1`.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::PolyOverZq;
+    /// use qfall_math::integer_mod_q::MatPolynomialRingZq;
+    /// use qfall_math::integer_mod_q::ModulusPolynomialRingZq;
+    /// use std::str::FromStr;
+    ///
+    /// let poly_mod = PolyOverZq::from_str("3  1 0 1 mod 17").unwrap();
+    /// let modulus = ModulusPolynomialRingZq::try_from(&poly_mod).unwrap();
+    ///
+    /// let matrix = MatPolynomialRingZq::new(5, 10, modulus).unwrap();
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type
+    /// [`InvalidMatrix`](MathError::InvalidMatrix)
+    /// if the number of rows or columns is `0`.
+    /// - Returns a [`MathError`] of type [`OutOfBounds`](MathError::OutOfBounds)
+    /// if the number of rows or columns is negative or it does not fit into an [`i64`].
+    #[allow(dead_code)]
+    pub fn new(
+        num_rows: impl TryInto<i64> + Display + Copy,
+        num_cols: impl TryInto<i64> + Display + Copy,
+        modulus: ModulusPolynomialRingZq,
```

Comment:
Better take in the modulus only as a reference and use clone internally, as it is a RC